dek-registry/src/main/java/io/confluent/dekregistry/storage/DefaultDekCacheUpdateHandler.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.dekregistry.storage;
+
+import java.io.IOException;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultDekCacheUpdateHandler implements DekCacheUpdateHandler {
+
+  private static final Logger log = LoggerFactory.getLogger(DefaultDekCacheUpdateHandler.class);
+
+  private final DekRegistry dekRegistry;
+
+  public DefaultDekCacheUpdateHandler(DekRegistry dekRegistry) {
+    this.dekRegistry = dekRegistry;
+  }
+
+  @Override
+  public ValidationStatus validateUpdate(
+      EncryptionKeyId key,
+      EncryptionKey value,
+      TopicPartition tp,
+      long offset,
+      long timestamp) {
+    if (value != null) {
+      // Store the offset and timestamp in the cached value
+      value.setOffset(offset);
+      value.setTimestamp(timestamp);
+    }
+
+    return ValidationStatus.SUCCESS;
+  }
+
+  /**
+   * Invoked on every new schema written to the Kafka store
+   *
+   * @param key   Key associated with the schema.
+   * @param value Value written to the Kafka lookupCache
+   */
+  @Override
+  public void handleUpdate(
+      EncryptionKeyId key,
+      EncryptionKey value,
+      EncryptionKey oldValue,
+      TopicPartition tp,
+      long offset,
+      long timestamp) {
+    String tenant = key.getTenant();
+    if (value == null) {
+      if (oldValue != null) {
+        dekRegistry.getMetricsManager().decrementKeyCount(tenant, key.getType());
+      }
+    } else if (oldValue == null) {
+      dekRegistry.getMetricsManager().incrementKeyCount(tenant, key.getType());
+    }
```

Comment:
for my own understanding, what exactly is happening here?

Reply:
We're incrementing a metrics for key count in case of adding to the cache, and decrementing in case of removing from the cache (removal involves a tombstone)